### PR TITLE
refactor(core): move flag parsing to bootstrap

### LINKED ISSUE #936: Use debug flag before cobra prerun
Currently, –debug flag is implemented with cobra persistent flags. As a result, the debug logger is not available before the prerun step. It could be great to directly use pflag and move the [debug logger init](https://github.com/scaleway/scaleway-cli/blob/v2/internal/core/cobra_utils.go#L118-L122) in bootstrap function.

Proposition:

- Add debug bool in BootStrapConfig. This flag is inited in main based on -D flag.
- We should keep cobra just for usage documentation (add a code comment to document that)
- Make sure the autocomplete -- notation still works.
- Add for instance a log debug message in bootstrap to see if the debug are showing up
- When the scoped logger is implemented we should be able to create a new logger in the Bootstrap function ad make sure it is accessible everywhere using context

### PR DESCRIPTION
**Pull request by jerome-quere**
relates to #964
Fixes #936

Blocked by https://github.com/scaleway/scaleway-sdk-go/pull/395

### CODE REVIEW

File: internal/core/context.go [hunk 1]
```diff
@@ -36,9 +36,9 @@ const (
 	metaContextKey contextKey = iota
 )
 
-// newMetaContext creates a new ctx with injected meta and returns it.
-func newMetaContext(meta *meta) context.Context {
-	return context.WithValue(context.Background(), metaContextKey, meta)
+// injectMeta creates a new ctx with injected meta and returns it.
```
Comment (Sh4d1) [inline internal/core/context.go:39]: nit: modifies the given context

Comment (jerome-quere) [reply internal/core/context.go:39]: Context are imutable object in Go. `WithValue` create a new child context of the on given as parameter

Comment (Sh4d1) [reply internal/core/context.go:39]: Yep but it's based on the context passed to the function

Comment (Sh4d1) [reply internal/core/context.go:39]: Something like `creates a new ctx based on the given one in the arguments`

File: internal/core/client.go [hunk 2]
```diff
@@ -96,81 +94,94 @@ More info: https://github.com/scaleway/scaleway-sdk-go/tree/master/scw#scaleway-
 }
 
 // validateProfile validate the final profile
```
Comment (Sh4d1) [inline internal/core/client.go:96]: Same nit for the comment

File: internal/core/build_info_test.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 var fakeCommand = &Command{
 	Namespace: "plop",
 	ArgsType:  reflect.TypeOf(args.RawArgs{}),
+	NoClient:  true,
```
Comment (remyleone) [inline internal/core/build_info_test.go:17]: Should we use a term such as `AllowAnonymousClient` ?

Comment (jerome-quere) [reply internal/core/build_info_test.go:17]: tracked in #968

**Pull request merged**